### ISSUE TITLE
"Module has no attribute" false positive

### ISSUE THREAD

**Issue by danieleades** [python/mypy#12921]
cross post from https://github.com/swyddfa/esbonio/issues/399

I'm seeing some type errors I can't explain

I have the following code-

```python
from esbonio.lsp import LanguageFeature
from esbonio.lsp.rst import CompletionContext, DefinitionContext, HoverContext
from esbonio.lsp.sphinx import SphinxLanguageServer
```

which is producing the following mypy errors-

```
sphinx_needs/lsp/esbonio.py:9: error: Module "esbonio.lsp" has no attribute "LanguageFeature"  [attr-defined]
sphinx_needs/lsp/esbonio.py:10: error: Module "esbonio.lsp.rst" has no attribute "CompletionContext"  [attr-defined]
sphinx_needs/lsp/esbonio.py:10: error: Module "esbonio.lsp.rst" has no attribute "DefinitionContext"  [attr-defined]
sphinx_needs/lsp/esbonio.py:10: error: Module "esbonio.lsp.rst" has no attribute "HoverContext"  [attr-defined]
sphinx_needs/lsp/esbonio.py:11: error: Module "esbonio.lsp.sphinx" has no attribute "SphinxLanguageServer"  [attr-defined]
sphinx_needs/lsp/esbonio.py:25: error: Class cannot subclass "LanguageFeature" (has type "Any")  [misc]
```

I can see that `esbonio.lsp` clearly *does* export these objects, so where is the issue coming from?

the code can be found here - https://github.com/useblocks/sphinxcontrib-needs/pull/584

esbonio version: 0.13.0

**Comment by danieleades** [comment]
> Do you have `--namespace-packages`?

Negative. Should I?

**Comment by hauntsaninja** [comment]
Yeah, could you try?

**Comment by danieleades** [comment]
That works a treat!

I'll close this issue.

Any downsides or gotchas to adding this configuration option?
